src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1855,7 +1888,7 @@ impl Cage {
                 Socket(socket_filedesc_obj) => {
                     let sock_tmp = socket_filedesc_obj.handle.clone();
                     let sockhandle = sock_tmp.write();
-
+                    // Check the domain of the socket (IPv4 or IPv6)
                     match sockhandle.domain {
                         AF_INET | AF_INET6 => match sockhandle.protocol {
                             IPPROTO_TCP => {
```

Comment:
Could you add more comments - because the part after line 1895 seems a bit confusing